error/s2n_errno.h

```diff
@@ -191,6 +191,7 @@ typedef enum {
     S2N_ERR_FREE_STATIC_BLOB,
     S2N_ERR_RESIZE_STATIC_BLOB,
     S2N_ERR_NO_AVAILABLE_BORINGSSL_API,
```

Comment:
As `S2N_ERR_NO_SUPPORTED_LIBCRYPTO_API` replaced `S2N_ERR_NO_AVAILABLE_BORINGSSL_API`. Do we still need it in the list?

Reply:
From offline discussion: this error code is required else-where for `x25519` support. I suggest being consistent with naming in that case:  `S2N_ERR_NO_SUPPORTED_LIBCRYPTO_API` and `S2N_ERR_NO_SUPPORTED_BORINGSSL_API`